airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -181,4 +194,34 @@ def propagate_types_and_parameters(
 
     @staticmethod
     def _is_json_schema_object(propagated_component: Mapping[str, Any]) -> bool:
-        return propagated_component.get("type") == "object"
+        return propagated_component.get("type") == "object" or propagated_component.get("type") == [
+            "null",
+            "object",
+        ]
+
+    @staticmethod
+    def _has_nested_components(propagated_component: Mapping[str, Any]) -> bool:
+        has_nested_components = False
+        for k, v in propagated_component.items():
+            if isinstance(v, dict) and v.get("type"):
+                has_nested_components = True
+        return has_nested_components
+
+    def _process_nested_components(
+        self,
+        propagated_component: Dict[str, Any],
+        parent_field_identifier: str,
+        current_parameters: Mapping[str, Any],
+        use_parent_parameters: Optional[bool] = None,
+    ) -> Dict[str, Any]:
+        for k, v in propagated_component.items():
```

Comment:
nit: for convention we usually try to avoid single letter variables and describe them. It's probably more descriptive to make this `field_name` and `value`.